File: HomeWorks/src/Lesson6/Legs/ToshibaLeg.java
package Lesson6.Legs;

public class ToshibaLeg implements ILeg {
    private int price;

    public ToshibaLeg() {}

    public ToshibaLeg (int price) {
        this.price = price;
    }

    @Override
    public void step() {
        System.out.println("Шагает нога Toshiba");
    }

    public void setPrice(int price) {
        this.price = price;
    }

    @Override
    public int getPrice() {
        return price;
    }
}

File: HomeWorkKotlin/src/Robot/Legs/ToshibaLeg.kt
package Robot.Legs

class ToshibaLeg(private val price_param : Int) : ILeg {

    private var price = price_param
        set(value) {
            if(value > 0) field = value
        }

    override fun step() {
        println("Шагает нога Toshiba")
    }

    override fun getLegPrice(): Int = price
}